### ISSUE TITLE
Monitors should be displayed as actual size and respect scaling, this will prevent strange behavior

### ISSUE THREAD

**Issue by cmot17** [bitgapp/Cursr#4]
### Describe the Bug/Issue

I have 3 27" monitors. The center one is 2560x1440, and the left and right ones are 3840x2160. The center monitor is slightly raised relative to the others, something like this:
![image](https://user-images.githubusercontent.com/25753364/125550724-67f25c97-1b34-4bed-8647-bbdcba66aa60.png)

However, windows handles this poorly and it shows up like this in display settings:
![image](https://user-images.githubusercontent.com/25753364/125550819-6f8195e9-78dd-4b6b-a09a-e6ee9caf57f4.png)

With the center monitor set to 100% scaling and the side monitors set to 150%, UI elements are physically the same size. However, Cursr seems to handle this poorly. For instance, if I set up my monitors like this:
![image](https://user-images.githubusercontent.com/25753364/125551039-dfc74f0e-0bdd-4ee0-8abd-04ca8f5e5ff3.png)
You can see that my mouse is (according to Cursr) off the screen. In reality, it was just near the top middle of my right screen.
If I move my monitors in Windows settings so that they are configured like this:
![image](https://user-images.githubusercontent.com/25753364/125551225-8a50505c-7979-481d-9801-02eed3d887a3.png)
Cursr then works mostly as intended. However, it assumes that my monitors are perfectly aligned, still resulting in my mouse jumping ~1 inch each time it transitions between monitors.
### Expected Behavior

It would be very helpful to have monitors displayed in actual size in Cursr, and to be able to move them around within the app to adjust transitions.

### Your Environment
 - OS details: Windows 10 Pro 20H1
 - Cursr Version 1.2.0

**Comment by ragauskl** [comment]
Noted, thanks for reporting that, I have couple things left for release of 1.3.0 and I'll look into this for version 1.4.0

**Comment by ragauskl** [comment]
@cmot17 Is the off the screen cursor issue still happening for you in latest version 1.3.3?

As for drawing monitors as actual size, I've added it to 1.4.0 as option, so it will be there once 1.4.0 is complete, in ~ first half of autumn

**Comment by ragauskl** [comment]
I had to exclude the feature from 1.4.0 (which is live now) due to some issues, but I will be looking into adding it back in soon, in nearest minor version updates

**Comment by cmot17** [comment]
Hi,
Thanks for all the updates.
It seems like the issues with the cursor going off-screen have been completely resolved for me. 
I still can't adjust the size of the monitors, which makes the transitions basically unusable.

Also, I really hope you will reconsider the subscription model. I know there will be a permanent license, but I find a subscription model to be extremely predatory in software with no recurring server costs. Especially $3 a month, when there are multiple free alternatives.

**Comment by ragauskl** [comment]
# Quick Summary

Monitor resizing - Maybe you're doing something wrong or expect Cursr to work in a way that it's not built to, with latest (but paid) feature of splitting borders into segments to pixel perfect size you can setup transitions to cover 90% of use cases. 

Also Windows and Cursr are drawing monitors correctly and exactly as computer sees the monitors - what you'd like to see is nicer from user perspective ( and is coming out with next feature - KVM - release )  but it's not theoretically correct and does require a decent amount of additional work/calculations to be done.

Pricing - I'm not an expert in this field and try to find a balance in decisions between information from credible sources and listening to customer feedback,  I definitely understand why most people don't like subscriptions, but I had many reasons to start from subscription model  (listed below) and will be keeping it, alongside 2 tiers (advanced and expert feature sets) of Perpetual Pro license.

Cursr does actually have reoccurring server costs, but not only 'server' costs count as reoccurring.

---


> Full answer is kinda long , so in case it's too long I added summary above ^ , but if you're interested feel free to read the full answer below:


---

# Monitor resizing and moving

Regarding the size of monitors displayed in Cursr, if you can't setup transitions to work there is a chance you're doing something wrong or have different expectations of how Cursr should work compared to how it does and is built to work. With the 'resizable borders' feature in Pro version you can setup transitions to be pixel perfect and split/merge border into segments to achieve various transitions. I do understand that you don't like the subscription model, but just to see if it would work for you - you can use free demo mode, it does have a cooldown however, alternative is to wait for new Cursr features and/or perpetual license (permanent license with some limitations/conditions).

Reason why the resizing/moving feature you've asked about had to be delayed is that Cursr now uses simplest solution - draws displays in the correct physical size in pixels, that's how your computer actually sees the monitors, no matter the scale or physical size which is what the user sees. That's why Windows draws is like that as well because technically that's the correct way. Where's the resizing and adjusting position to match what user sees is not theoretically correct. It is what you see as a user but not how computer actually works under the hood.

I'm not the best at giving examples (so let me know if this makes no sense) but I'll try to provide one to explain how Cursr works now vs what the feature requires :
        Currently Cursr can draw everything on a single (x, y) coordinate system with exact monitor size and mouse pointer position, without having to do any additional computations. In order to allow for resizing and moving monitors to match what you see in real life, Cursr needs to draw every single monitor on it's own coordinate system (to allow moving and scaling) of different scales (to allow resizing) and overlay them on same background/canvas. This also requires position of drawn in-app mouse pointer to be transformed between the different coordinates systems based on which screen it is on.

So such solution requires more calculations to be done and creates more place for bugs (but nevertheless it's needed both because you as a user would like to use it and because it's needed for another feature - KVM - to work).

# Pricing

I believe I understand your viewpoint and see why people see subscriptions as predatory, but almost anything can be predatory in certain circumstances, permanent licenses can also be predatory when they're overpriced for the product. And if no one else does a certain pricing model for same product doesn't always mean you shouldn't or can't. Let me explain my reasoning for subscription and further licensing plans: 

My initial reason for starting of subscription was that it's most forgiving in terms of development time (personally for me) and bug fixing and both are critical to Cursr as it's still in early stages of development:
* I already had some clue/pinch of experience working with it subscription model, besides that small experience I had none other of creating/managing payment system and licensing. So working with subscription was faster to implement for me
* Subscription model allowed me to issue licenses only for a month at a time, as I was still new to licensing systems, in worst case scenario bug where people can abuse system to issue many licenses for free, I would not loose much as they would get invalidated after a month, where's if it was permanent licenses I would have a much harder time fixing issues. As I learn more I can implement a permanent license with more confidence it won't be abused.
* Cursr currently does not have high reoccurring costs, but it's not a guarantee there won't be any. Currently there are some features planned/thought about that cannot be added without increasing reoccurring costs like cloud computing, but they are of low priority if possible at all. Again - subscription model was an easy 'cover all use cases in one model' pricing.
* Subscription model is basically "Get all available perks and features of Cursr for 3$/month", this is the easiest model to deal with as it covers all features current and any I'll be adding in future no matter of their complexity, therefore once it's released I don't need to worry about creating new types of licenses for different  usage scopes of the application. Which brings me to next section - future licensing plans:

## Future payment/licensing options

I haven't fully committed to specific prices, but currently plan is to have these options (note that these are for feature access, all bug fixes are free forever):
* Free - current free features will stay free forever
* Subscription 3$/month - All current and future features and perks available without any additional costs
* Perpetual Pro-Advanced <15$/license -  All current 'Pro' features except for KVM related ones are available forever- as in all features usable on single Operating System/computer (no display limit) + guaranteed/dedicated support for 1year after purchase
* Perpetual Pro-Expert>15$ (due to nature of KVM feature, first purchase will include 2 licenses) + ?$ (reduced price for every additional license) - All current 'Pro' features including KVM available forever  + guaranteed/dedicated support for 1year  after purchase

> Note that 'perpetual' licenses say 'current' features - this means that what you buy at the time will always be available to you. However new features that come out after the purchase might have to be purchased separately, depending on complexity of the feature, if it's easy - you'll likely get it for free if you already have a license, but in case the feature is really complex it would likely be sold as add-on at reduced price OR  you would see increase in perpetual license cost for new customers, and existing customer would have their existing license working and could just pay the difference for new features.
> Definition of 'forever' in terms of feature releases and bug fixes means that 'if they are released' in case where Cursr project is abandoned (hopefully it won't get to that) there won't be any new features or bug fixes until and if the project is resumed
>Just to make it clear - I currently have no plans for such complex features besides KVM which is already taken into account, but it could happen if I'll keep working on Cursr. It's not an exact plan but a rough idea from thinking ahead about what would happen in such case.

## Reoccurring costs

Cursr does indeed have reoccurring server costs (not one of reasons why subscriptions was added, but still a fact), it's partially for handling subscription pricing itself, however even without it a database and a server are required to securely handle user data for processing and keeping track of valid permanent licenses , both have to be live 24/7 and scalable which means there's a literal reoccurring cost. Alternative would be not to track licenses and users, just issue a license but that would come with it's own cons, like no option or complicated way to reset or recover a paid for license.

There are also other reoccurring costs:
* maintaining the application is a reoccurring cost
* so is customer support - especially when an issue is happening just to one or very small amount of users, in some (but not all) cases because of their issue that is not related to the application. Which is a cost of time (and time is money) for helping resolve the issue even if it wasn't caused by or related to the application
* also computers and environments within them change, and what is working now might not work in a year or two because Windows or other company decided to change some details about how computer works, this is a constant issue, and you can't foresee how much of time solving these will take away from you
* also as the user base grows - more support issues/emails come in - at some point you need to hire a dedicated person, that's a big reoccurring cost, and subscription does include dedicated support (for now it's just me but as user base grows it's inevitable that either I need to start hiring people or abandon project because it's out of  physical possibilities of one person to deal with)

## Other reasons for pricing

This pricing is not made to 'pray' on customers and pull out all the money, the project itself started because those mentioned 'free' alternatives were not working for me as I would like them to, I did not like them and I decided to do my own thing first for  fun, later I decided to quit my job and commit to this project full-time, see what I can learn from it and how far can I make it go, for how many people can I create something they will find useful and/or helpful.

I've been working on this for bit over a year now, I'm working on this alone and this project is not funded by anyone besides my personal savings with no additional income until recently, I've downgraded my lifestyle a lot to bare minimum be able to allow myself to just work on Cursr full-time and it's getting closer and closer to two options - make it profitable or abandon the project (just like the free alternatives you mentioned - they're free, but they're also abandoned or rarely maintained), free or cheap projects can be nice to users but they don't pay the bills and everyone has them.

I'm not saying that to complain, but just to explain that I'm trying to balance between what's good for the projects sustainability (according to creditable resources and other people with experience in the field) and  trying to keep customers happy by listening to feedback and doing what I can with rational reasoning.

Besides what I explained above, let me add that at least in case of Cursr I don't think users should keep buying subscription forever, at least personally I see need for subscription because:
* Someone might only need full Cursr functionality for a month in  a year, why pay for permanent license when you can just pay 3$ for that month? Talking from personal experience, before working on Cursr I only needed it's upcoming KVM feature once for a month, two tops, and never again needed it, the free version is enough for my personal needs. If there's one person like that, there's gonna be plenty more
* Maybe someone want's a longer 'trial' time without committing to a  permanent license
* Maybe someone likes it so much they just want to support the projects development in a way that's more sustainable to the project for some time

**Comment by ragauskl** [comment]
@cmot17 Displays are now draggable/movable in v1.5.0, however rendering the displays as 'actual' size was not requested by anyone else and there is list of higher priority features and issues. This is still on the to-do list but it might be some time before this is added to Cursr unless there will be more users interested in such functionality
